internal/store/engine_state.go

```diff
@@ -79,6 +80,10 @@ type EngineState struct {
 
 	// from GitHub
 	LatestTiltBuild model.TiltBuild
+
+	// Analytics Info
+	AnalyticsOpt        analytics.Opt // TODO(maia): remember to update this when user opts in/out
+	NeedsAnalyticsNudge bool          // TODO(maia): maybe this can convey more info, e.g. "there was error in /analytics/opt"
```

Comment:
i don't understand why you need this state at all?

afaict, the "source of truth" is "have any builds successfully completed yet?"

but you can infer this from the existing engine state. Creating a new state variable makes it more likely that we'll introduce bugs, because the reducer will change and you'll forget to keep this up to date.

it'd be simpler and less buggy to have a function that takes an EngineState and returns if you need an analytics nudge.

does that make sense?